### PR TITLE
osv updater

### PR DESCRIPTION
**Pull request by hdonnay**
Signed-off-by: Hank Donnay <hdonnay@redhat.com>

### CODE REVIEW

File: updater/osv/osv.go [hunk 1]
```diff
@@ -0,0 +1,801 @@
...
+	}
+	if imp > 0 { // Score is 0 when impact is 0 or below.
+		exp := 8.22 * ns[ /*AV*/ 0] * ns[ /*AC*/ 1] * ns[ /*PR*/ 2] * ns[ /*UI*/ 3]
+		s := exp + imp
+		if changed {
+			s *= 1.08
+		}
+		s = math.Min(s, 10)
+		// Roundup function, as spec'd.
+		i := int(s * 100_000)
```
Comment (crozzy) [inline updater/osv/osv.go:769]: didn't know (or had possibly forgotten) you could declare numerics like this, v.nice

File: updater/osv/osv.go [hunk 2]
```diff
@@ -0,0 +1,801 @@
...
+	r := make([]*claircore.Vulnerability, len(e.Vulnerability))
+	for i := range e.Vulnerability {
+		r[i] = &e.Vulnerability[i]
+	}
+	return r
+}
+
+// FromCVSS is an attempt at an implementation of the scale and formulas
+// described here: https://www.first.org/cvss/v3.1/specification-document#Qualitative-Severity-Rating-Scale
+func fromCVSS(s string) (sev claircore.Severity, err error) {
```
Comment (crozzy) [inline updater/osv/osv.go:651]: out of interest, did you see this: https://github.com/goark/go-cvss? Might be interesting to run [this test suite](https://github.com/goark/go-cvss/blob/master/v3/base/base_test.go) through this function, specifically:

```		{vector: "CVSS:3.1/AV:P/AC:H/PR:H/UI:R/S:U/C:N/I:N/A:N", score: 0.0, severity: SeverityNone},     //Zero metrics
		{vector: "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N", score: 7.5, severity: SeverityHigh},     //CVE-2015-8252
		{vector: "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:C/C:L/I:L/A:N", score: 6.1, severity: SeverityMedium},   //CVE-2013-1937
		{vector: "CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:C/C:L/I:L/A:N", score: 6.4, severity: SeverityMedium},   //CVE-2013-0375
		{vector: "CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:U/C:L/I:N/A:N", score: 3.1, severity: SeverityLow},      //CVE-2014-3566
		{vector: "CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:C/C:H/I:H/A:H", score: 9.9, severity: SeverityCritical}, //CVE-2012-1516
		{vector: "CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H", score: 8.8, severity: SeverityHigh},     //CVE-2012-0384
		{vector: "CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H", score: 7.8, severity: SeverityHigh},     //CVE-2015-1098
		{vector: "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N", score: 7.5, severity: SeverityHigh},     //CVE-2014-0160
		{vector: "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H", score: 9.8, severity: SeverityCritical}, //CVE-2014-6271
		{vector: "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:C/C:N/I:H/A:N", score: 6.8, severity: SeverityMedium},   //CVE-2008-1447
		{vector: "CVSS:3.1/AV:P/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H", score: 6.8, severity: SeverityMedium},   //CVE-2014-2005
		{vector: "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:C/C:L/I:N/A:N", score: 5.8, severity: SeverityMedium},   //CVE-2010-0467
		{vector: "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:C/C:N/I:L/A:N", score: 5.8, severity: SeverityMedium},   //CVE-2012-1342
		{vector: "CVSS:3.1/AV:N/AC:L/PR:L/UI:R/S:C/C:L/I:L/A:N", score: 5.4, severity: SeverityMedium},   //CVE-2014-9253
		{vector: "CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H", score: 7.8, severity: SeverityHigh},     //CVE-2009-0658
		{vector: "CVSS:3.1/AV:A/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H", score: 8.8, severity: SeverityHigh},     //CVE-2011-1265
		{vector: "CVSS:3.1/AV:P/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:N", score: 4.6, severity: SeverityMedium},   //CVE-2014-2019
		{vector: "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H", score: 8.8, severity: SeverityHigh},     //CVE-2015-0970
		{vector: "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:N", score: 7.4, severity: SeverityHigh},     //CVE-2014-0224
		{vector: "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:C/C:H/I:H/A:H", score: 9.6, severity: SeverityCritical}, //CVE-2012-5376
```
Just a thought

**Review by crozzy — COMMENTED** [review]
This looks very nice, one thing that would be nice to know is the DB size difference after the change, just to get a headstart on any potential issues with storage.

File: updater/osv/osv.go [hunk 3]
```diff
@@ -0,0 +1,801 @@
...
+		e.Package[i].Name = name
+		e.Package[i].Version = ver
+		e.pkgindex[key] = i
+	}
+	return &e.Package[i], ok
+}
+
+func (e *ecs) LookupRepository(name string) (r *claircore.Repository) {
+	e.hash.Reset()
+	fmt.Fprint(e.hash, name) // documented to never error
```
[the remark this answers is not in the document]
Comment (hdonnay) [reply updater/osv/osv.go:601]: oh, that doesn't need to be done, it could just be a string, yeah.

**Review by crozzy — APPROVED** [review]

**Pull request merged**